quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json

```diff
@@ -0,0 +1,141 @@
+      }
+    },
+    "skuName": {
+      "type": "string",
+      "defaultValue": "payShield10K_LMK1_CPS60",
+      "metadata": {
+        "description": "PayShield SKU name. It must be one of the following: payShield10K_LMK1_CPS60, payShield10K_LMK1_CPS250, payShield10K_LMK1_CPS2500, payShield10K_LMK2_CPS60, payShield10K_LMK2_CPS250, payShield10K_LMK2_CPS2500"
+      }
+    },
+    "vnetName": {
+      "type": "string",
+      "metadata": {
+        "description": "Virtual network name"
+      }
+    },
+    "vnetAddressPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "Virtual network address prefix"
+      }
+    },
+    "hsmSubnetName": {
+      "type": "String",
+      "metadata": {
+        "description": "Subnet name"
+      }
+    },
+    "hsmSubnetPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "Subnet prefix"
+      }
+    }
+  },
+  "variables": {},
+  "resources": [
+   {
+     "type": "Microsoft.HardwareSecurityModules/dedicatedHSMs",
+     "apiVersion": "2021-11-30",
+     "name": "[parameters('resourceName')]",
+	   "location": "[parameters('location')]",
+     "dependsOn": [
+      "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnetName'), parameters('hsmSubnetName'))]"
+     ],
+     "sku": {
+       "name": "[parameters('skuName')]"
+     },
+     "properties": {
+       "networkProfile": {
+         "subnet": {
+           "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnetName'), parameters('hsmSubnetName'))]"
+         }
+        },
+		"managementNetworkProfile": {
+          "subnet": {
+            "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnetName'), parameters('hsmSubnetName'))]"
+          }
+        },
+        "stampId": "[parameters('stampId')]"
+     }
+   },
+    {
+      "type": "Microsoft.Network/virtualNetworks",
+      "apiVersion": "2020-11-01",
+      "name": "[parameters('vnetName')]",
+      "location": "[parameters('location')]",
+      "tags": {
+        "fastpathenabled": "true"
+      },
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": [
+            "[parameters('vnetAddressPrefix')]"
+          ]
+        },
+        "subnets": [
+          {
+            "name": "[parameters('hsmSubnetName')]",
+            "properties": {
+              "addressPrefix": "[parameters('hsmSubnetPrefix')]",
+              "delegations": [
+                {
+                  "name": "Microsoft.HardwareSecurityModules.dedicatedHSMs",
+                  "properties": {
+                    "serviceName": "Microsoft.HardwareSecurityModules/dedicatedHSMs"
+                  }
+                }
+              ],
+              "privateEndpointNetworkPolicies": "Enabled",
+              "privateLinkServiceNetworkPolicies": "Enabled"
+            }
+          }
+        ],
+        "enableDdosProtection": false
+      }
+    },
+    {

```

Comment:
why is this defined twice in the template?

Reply:
enableDdosProtection? I only see it once.

Reply:
The hsmSubnet is defined on the vnet resource as an array element and also as a standalone child resource...